tests/Api.Integration.Tests/RegisterTests.cs

```diff
@@ -0,0 +1,203 @@
+using System.Net;
+using System.Net.Http.Json;
+using System.Text.Json;
+using Bogus;
+using Fido2NetLib;
+using FluentAssertions;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.VirtualAuth;
+using Passwordless.Api.Endpoints;
+using Passwordless.Service.Models;
+using Xunit;
+
+namespace Passwordless.Api.Integration.Tests;
+
+public class RegisterTests : IClassFixture<PasswordlessApiFactory>
+{
+    private readonly HttpClient _client;
+
+    private static readonly Faker<RegisterToken> TokenGenerator = new Faker<RegisterToken>()
+        .RuleFor(x => x.UserId, Guid.NewGuid().ToString())
+        .RuleFor(x => x.DisplayName, x => x.Person.FullName)
+        .RuleFor(x => x.Username, x => x.Person.Email)
+        .RuleFor(x => x.Attestation, "None")
+        .RuleFor(x => x.Discoverable, true)
+        .RuleFor(x => x.UserVerification, "Preferred")
+        .RuleFor(x => x.Aliases, x => new HashSet<string> { x.Person.FirstName })
+        .RuleFor(x => x.AliasHashing, false)
+        .RuleFor(x => x.ExpiresAt, DateTime.UtcNow.AddDays(1))
+        .RuleFor(x => x.TokenId, Guid.Empty);
+
+    public RegisterTests(PasswordlessApiFactory apiFactory)
+    {
+        _client = apiFactory.CreateClient();
+        _client.DefaultRequestHeaders.Add("User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36");
+    }
+
+    [Fact]
+    public async Task Server_Can_Successfully_Retrieve_Token_For_Registration()
+    {
+        var request = TokenGenerator.Generate();
+
+        var response = await _client.AddSecretKey().PostAsJsonAsync("/register/token", request);
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        var registerTokenResponse = await response.Content.ReadFromJsonAsync<RegisterEndpoints.RegisterTokenResponse>();
+        registerTokenResponse.Should().NotBeNull();
+        registerTokenResponse!.Token.Should().StartWith("register_");
+    }
+
+    [Fact]
+    public async Task Client_Can_Begin_Registration_Of_User_Credential_With_Server()
+    {
+        var tokenRequest = TokenGenerator.Generate();
+        var tokenResponse = await _client.AddSecretKey().PostAsJsonAsync("/register/token", tokenRequest);
```

Comment:
Maybe the client should be generated per-test (instead of shared in the class), so that way you don't accidentally retain the old `SecretKey`/`ManagementKey` value?

Kinda similar to this: https://github.com/passwordless/passwordless-dotnet/blob/24661e9bbc064b6ecfae8de7a31393655517d170/tests/Passwordless.Tests/TokenTests.cs#L20

Reply:
If I remember correctly, the constructor of the class will run for each fact. So each of them is getting a new client and isn't shared across the different facts.